sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -459,5 +495,54 @@ internal static bool GlobalDisableAutoRecording
                 return disableAutoRecording || GlobalIsRunningInCI;
             }
         }
+
+        private void BootStrapTestResources()
+        {
+            lock (s_syncLock)
+            {
+                try
+                {
+                    if (!s_isWindows ||
+                        s_bootstrappingAttemptedTypes.Contains(_type) ||
+                        Mode == RecordedTestMode.Playback ||
+                        GlobalIsRunningInCI)
```

Comment:
Nit: I'd reorder these with faster checks first - basically, reverse these last three.

Reply:
GlobalIsRunningInCI does an environment variable lookup whenever invoked. 
I know these are cached for the process, but isn't it still a relatively expensive system call?